File: src/main/kotlin/top_75/_700_searchina_binary_search_tree.java
package top_75;

public class _700_searchina_binary_search_tree {
    public TreeNode searchBST(TreeNode root, int val) {
        if (root == null) return null;
        if (root.val == val) return root;
        var left = searchBST(root.left, val);
        var right = searchBST(root.right, val);
        return left == null ? right : left;
    }

    public static void main(String[] args) {
        var s = new _700_searchina_binary_search_tree();
        s.searchBST(TreeNode.buildTree(4, 2, 7, 1, 3), 2).printTree();
    }
}

File: src/main/kotlin/topic/tree/easy/_700_searchina_binary_search_tree.kt
package topic.tree.easy

import common.TreeNode

class _700_searchina_binary_search_tree {
    fun searchBST(root: TreeNode?, `val`: Int): TreeNode? {
        root ?: return null
        if (root.`val` == `val`) return root
        return searchBST(root.left, `val`) ?: searchBST(root.right, `val`)
    }
}
